service/windows/src/node.rs

```diff
@@ -0,0 +1,478 @@
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::iter::FromIterator;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::os::windows::process::CommandExt;
+use std::path::PathBuf;
+use std::process::{Child, Command, ExitStatus, Stdio};
+use tempfile::{NamedTempFile, TempPath};
+use toml::Value;
+use winapi::um::winbase::CREATE_NEW_PROCESS_GROUP;
+
+/// Default RPC port
+const DEFAULT_RPC_PORT: u16 = 10000;
+
+pub struct LogFileConfig {
+    pub path: PathBuf,
+    pub roll_size: Option<String>,
+    pub roll_count: Option<u16>,
+    pub roll_pattern: Option<String>,
+}
+
+pub enum LoggerConfig {
+    NoLog,
+    ConfigLog(PathBuf),
+    FileLog(LogFileConfig),
+}
+
+/// Runner for a node
+pub struct NodeConfig {
+    // Name of the node for logging purposes
+    pub name: String,
+    // Bootstrap node string
+    pub bootstrap_nodes: String,
+    // Configuration directory for the node
+    pub config_dir: PathBuf,
+    // Data directory for the node
+    pub data_dir: PathBuf,
+    // Credentials file (optional)
+    pub baker_credentials: Option<PathBuf>,
+    // Address to accept GRPC requests on
+    pub rpc_address: Option<IpAddr>,
+    // Port for GRPC requests
+    pub rpc_port: Option<u16>,
+    // Whether GRPC is enabled on the node
+    pub rpc_enabled: Option<bool>,
+    // GRPC authentication token
+    pub rpc_token: Option<String>,
+    // Address to listen for peer-to-peer connections on
+    pub listen_address: Option<IpAddr>,
+    // Port for peer-to-peer connections
+    pub listen_port: Option<u16>,
+    // Log configuration file; if specified, other log optsion
+    pub log_config: LoggerConfig,
+    // log level
+    pub log_level: Option<Level>,
+    // Node binary
+    pub node_bin: Option<PathBuf>,
+    // Node environment variables
+    pub node_env: HashMap<String, String>,
+    // Node arguments
+    pub node_args: Vec<String>,
+    // Collector enabled
+    pub collector_enabled: Option<bool>,
+    // Collector URL
+    pub collector_url: Option<String>,
+    // Collector node name
+    pub collector_node_name: Option<String>,
+    // Collector interval
+    pub collector_interval: Option<u64>,
+    // Collector logging file
+    pub collector_log_file: Option<PathBuf>,
+    // Collector binary
+    pub collector_bin: Option<PathBuf>,
+    // Collector environment variables
+    pub collector_env: HashMap<String, String>,
+    // Collector arguments
+    pub collector_args: Vec<String>,
+}
+
+/// A started node.
+pub struct Node {
+    /// The child process for the node.
+    node_process: Child,
+    /// The configuration settings used to start the node.
+    pub node_config: NodeConfig,
+    /// A temporary file used to redirect the node's stderr.
+    stderr_path: TempPath,
+    /// The child process for the collector.
+    collector_process: Option<Child>,
+    /// A temporary file used for configuring the node's logging.
+    /// (When this is dropped, the file will be deleted.)
+    #[allow(dead_code)]
```

Comment:
Why is this needed?

Reply:
The field is never explicitly read, but is kept so that it will be dropped when the `Node` is dropped. Will rename to `_log_config_file` so that the annotation is not required.